ArcGISRuntimeSDKQt_CppSamples/Layers/GroupLayers/GroupLayers.cpp

```diff
@@ -0,0 +1,121 @@
+  ArcGISSceneLayer* sceneLayer2 = new ArcGISSceneLayer(QUrl("https://scenesampleserverdev.arcgis.com/arcgis/rest/services/Hosted/DevA_Pathways/SceneServer/layers/0"), this);
+  ArcGISSceneLayer* sceneLayer3 = new ArcGISSceneLayer(QUrl("https://scenesampleserverdev.arcgis.com/arcgis/rest/services/Hosted/DevA_BuildingShell_Textured/SceneServer/layers/0"), this);
+  GroupLayer* groupLayer = new GroupLayer(QList<Layer*>{sceneLayer1, sceneLayer2, sceneLayer3}, this);
+  groupLayer->setName("Group: Dev A");
+
+  // Create 2 layers outside of group layer
+  ArcGISSceneLayer* sceneLayer4 = new ArcGISSceneLayer(QUrl("https://scenesampleserverdev.arcgis.com/arcgis/rest/services/Hosted/PlannedDemo_BuildingShell/SceneServer/layers/0"), this);
+  ServiceFeatureTable* ft = new ServiceFeatureTable(QUrl("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevelopmentProjectArea/FeatureServer/0"), this);
+  FeatureLayer* featureLayer = new FeatureLayer(ft, this);
+
+  // Add layers to the scene
+  m_scene->operationalLayers()->append(groupLayer);
+  m_scene->operationalLayers()->append(sceneLayer4);
+  m_scene->operationalLayers()->append(featureLayer);
+
+  connect(m_scene, &Scene::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    m_layerListModel = m_scene->operationalLayers();
+    emit layerListModelChanged();
+  });
+}
+
+GroupLayers::~GroupLayers() = default;
+
+void GroupLayers::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<GroupLayers>("Esri.Samples", 1, 0, "GroupLayersSample");
+  qmlRegisterUncreatableType<LayerListModel>("Esri.Samples", 1, 0, "AbstractListModel", "AbstractListModel is uncreateable");
+}
+
+SceneQuickView* GroupLayers::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void GroupLayers::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+  {
+    return;
+  }
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  // create initial viewpoint
+  Geometry env = Geometry::fromJson(R"({"spatialReference":{
+                                    "wkid":4326
+                                    },
+                                    "xmax":-122.67960721754773,
+                                    "xmin":-122.68647066116789,
+                                    "ymax":45.53584958588318,
+                                    "ymin":45.531539857343745
+                                    })");
+  // zoom to viewpoint
+  m_sceneView->setViewpoint(Viewpoint(env.extent()));
+
+  emit sceneViewChanged();
+}
+
+LayerListModel* GroupLayers::getGroupLayerListModel(int layerId)
+{
+  if (!m_scene)
+    return nullptr;
+
+  auto groupLayer = static_cast<GroupLayer*>(m_scene->operationalLayers()->at(layerId));
+
+  if (groupLayer)
+    return groupLayer->layers();
+  else
+    return nullptr;
```

Comment:
This block can be reformatted a bit for readability. IMO you don't need the `else` case.

```c++
  if (groupLayer)
    return groupLayer->layers();

  return nullptr;
```